test/utils/deployment.ts

```diff
@@ -0,0 +1,121 @@
+ * should be.
+ * @param shouldPrintApiLogs Whether or not the 0x-api logs should be surfaced.
+ * @param shouldPrintDependencyLogs Whether or not the 0x-api's dependencies
+ *        should surface their logs.
+ * TODO(jalextowle): It would be a good improvement to be able to specify log
+ * files where the logs should actually be written.
+ */
+export interface LoggingConfig {
+    shouldPrintApiLogs?: boolean;
+    shouldPrintDependencyLogs?: boolean;
+}
+
+/**
+ * Sets up a 0x-api instance.
+ * @param loggingConfig Whether or not the logs from the setup functions should
+ *        be printed.
+ */
+export async function setupApiAsync(loggingConfig: LoggingConfig = {}): Promise<void> {
+    if (yarnStartProcess) {
+        throw new Error('Old 0x-api instance has not been torn down');
+    }
+    await setupDependenciesAsync(loggingConfig.shouldPrintDependencyLogs || false);
+    yarnStartProcess = spawn('yarn', ['start'], {
+        cwd: apiRootDir,
+    });
+    if (loggingConfig.shouldPrintApiLogs) {
+        yarnStartProcess.stdout.on('data', chunk => {
+            neatlyPrintChunk('[0x-api]', chunk);
+        });
+        yarnStartProcess.stderr.on('data', chunk => {
+            neatlyPrintChunk('[0x-api | error]', chunk);
+        });
+    }
+    // Wait for the API to boot up
+    // HACK(jalextowle): This should really be replaced by log-scraping, but it
+    // does appear to work for now.
+    await sleepAsync(5000); // tslint:disable-line:custom-no-magic-numbers
+}
+
+/**
+ * Tears down the old 0x-api instance.
+ * @param loggingConfig Whether or not the logs from the teardown functions should
+ *        be printed.
+ */
+export async function teardownApiAsync(loggingConfig: LoggingConfig = {}): Promise<void> {
+    if (!yarnStartProcess) {
+        throw new Error('There is no 0x-api instance to tear down');
+    }
+    yarnStartProcess.kill();
+    await teardownDependenciesAsync(loggingConfig.shouldPrintDependencyLogs || false);
+}
+
+/**
+ * Sets up 0x-api's dependencies.
+ * @param shouldPrintLogs Whether or not the logs from `docker-compose up`
+ *        should be printed.
+ */
+export async function setupDependenciesAsync(shouldPrintLogs: boolean = false): Promise<void> {
+    const up = spawn('docker-compose', ['up'], {
+        cwd: apiRootDir,
+        env: {
+            ...process.env,
+            ETHEREUM_RPC_URL: 'http://ganache:8545',
+            ETHEREUM_CHAIN_ID: '1337',
+        },
+    });
+    if (shouldPrintLogs) {
+        up.stdout.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose up]', chunk);
+        });
+        up.stderr.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose up | error]', chunk);
+        });
+    }
+    // Wait for the dependencies to boot up.
+    // HACK(jalextowle): This should really be replaced by log-scraping, but it
+    // does appear to work for now.
+    await sleepAsync(10000); // tslint:disable-line:custom-no-magic-numbers
```

Comment:
Yikes, does it take that look to boot up? Would log scraping make this faster?

Reply:
That's the upper limit (mostly because of `docker pull`'s inconsistent timing on different runs. I think log scraping will make this considerably faster most of the time.